File: pkg/synnergy_network/core/wallet/walletSDK/GUIs/desktop/wallet_cryptoGui.py
import tkinter as tk
from tkinter import messagebox, simpledialog
import requests
import json

class WalletCryptoGUI:
    def __init__(self, root):
        self.root = root
        self.root.title("Wallet Crypto GUI")
        
        self.create_widgets()

    def create_widgets(self):
        tk.Button(self.root, text="Generate Key Pair", command=self.generate_key_pair).pack(pady=10)
        tk.Button(self.root, text="Encrypt Data", command=self.encrypt_data).pack(pady=10)
        tk.Button(self.root, text="Decrypt Data", command=self.decrypt_data).pack(pady=10)
        tk.Button(self.root, text="Sign Data", command=self.sign_data).pack(pady=10)
        tk.Button(self.root, text="Verify Signature", command=self.verify_signature).pack(pady=10)
        tk.Button(self.root, text="Hash Data", command=self.hash_data).pack(pady=10)
        
    def generate_key_pair(self):
        response = requests.post("http://localhost:8000/api/generate_keypair")
        if response.status_code == 200:
            data = response.json().get("data", {})
            messagebox.showinfo("Key Pair Generated", json.dumps(data, indent=4))
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to generate key pair"))

    def encrypt_data(self):
        data = simpledialog.askstring("Input", "Enter data to encrypt:")
        passphrase = simpledialog.askstring("Input", "Enter passphrase:")
        response = requests.post("http://localhost:8000/api/encrypt_data", json={"data": data, "passphrase": passphrase})
        if response.status_code == 200:
            encrypted_data = response.json().get("data", {}).get("encrypted_data", "")
            messagebox.showinfo("Encrypted Data", encrypted_data)
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to encrypt data"))

    def decrypt_data(self):
        data = simpledialog.askstring("Input", "Enter data to decrypt:")
        passphrase = simpledialog.askstring("Input", "Enter passphrase:")
        response = requests.post("http://localhost:8000/api/decrypt_data", json={"data": data, "passphrase": passphrase})
        if response.status_code == 200:
            decrypted_data = response.json().get("data", {}).get("decrypted_data", "")
            messagebox.showinfo("Decrypted Data", decrypted_data)
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to decrypt data"))

    def sign_data(self):
        data = simpledialog.askstring("Input", "Enter data to sign:")
        private_key = simpledialog.askstring("Input", "Enter private key:")
        response = requests.post("http://localhost:8000/api/sign_data", json={"data": data, "private_key": private_key})
        if response.status_code == 200:
            signature = response.json().get("data", {}).get("signature", "")
            messagebox.showinfo("Signature", signature)
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to sign data"))

    def verify_signature(self):
        data = simpledialog.askstring("Input", "Enter data:")
        public_key = simpledialog.askstring("Input", "Enter public key:")
        signature = simpledialog.askstring("Input", "Enter signature:")
        response = requests.post("http://localhost:8000/api/verify_signature", json={"data": data, "public_key": public_key, "signature": signature})
        if response.status_code == 200:
            is_valid = response.json().get("data", {}).get("is_valid", False)
            messagebox.showinfo("Verification Result", f"Signature is valid: {is_valid}")
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to verify signature"))

    def hash_data(self):
        data = simpledialog.askstring("Input", "Enter data to hash:")
        response = requests.post("http://localhost:8000/api/hash_data", json={"data": data})
        if response.status_code == 200:
            hash_value = response.json().get("data", {}).get("hash", "")
            messagebox.showinfo("Hash Value", hash_value)
        else:
            messagebox.showerror("Error", response.json().get("message", "Failed to hash data"))

if __name__ == "__main__":
    root = tk.Tk()
    app = WalletCryptoGUI(root)
    root.mainloop()

File: pkg/synnergy_network/core/wallet/walletSDK/GUIs/web/components/wallet_cryptoGui.js
import React, { useState } from 'react';
import {
  generateKeyPair,
  encryptData,
  decryptData,
  signData,
  verifySignature,
  hashData
} from '../api_service/wallet_crypto_api_service';

const WalletCryptoGui = () => {
  const [data, setData] = useState('');
  const [passphrase, setPassphrase] = useState('');
  const [privateKey, setPrivateKey] = useState('');
  const [publicKey, setPublicKey] = useState('');
  const [signature, setSignature] = useState('');
  const [result, setResult] = useState('');

  const handleGenerateKeyPair = async () => {
    try {
      const response = await generateKeyPair();
      setPrivateKey(response.private_key);
      setPublicKey(response.public_key);
      setResult('Key pair generated successfully');
    } catch (error) {
      setResult('Failed to generate key pair');
    }
  };

  const handleEncryptData = async () => {
    try {
      const encryptedData = await encryptData(data, passphrase);
      setResult(`Encrypted Data: ${encryptedData}`);
    } catch (error) {
      setResult('Failed to encrypt data');
    }
  };

  const handleDecryptData = async () => {
    try {
      const decryptedData = await decryptData(data, passphrase);
      setResult(`Decrypted Data: ${decryptedData}`);
    } catch (error) {
      setResult('Failed to decrypt data');
    }
  };

  const handleSignData = async () => {
    try {
      const sign = await signData(data, privateKey);
      setSignature(sign);
      setResult(`Signature: ${sign}`);
    } catch (error) {
      setResult('Failed to sign data');
    }
  };

  const handleVerifySignature = async () => {
    try {
      const isValid = await verifySignature(data, publicKey, signature);
      setResult(`Signature Valid: ${isValid}`);
    } catch (error) {
      setResult('Failed to verify signature');
    }
  };

  const handleHashData = async () => {
    try {
      const hash = await hashData(data);
      setResult(`Hash: ${hash}`);
    } catch (error) {
      setResult('Failed to hash data');
    }
  };

  return (
    <div>
      <h2>Wallet Crypto Operations</h2>
      <div>
        <label>Data:</label>
        <input
          type="text"
          value={data}
          onChange={(e) => setData(e.target.value)}
        />
      </div>
      <div>
        <label>Passphrase:</label>
        <input
          type="password"
          value={passphrase}
          onChange={(e) => setPassphrase(e.target.value)}
        />
      </div>
      <div>
        <label>Private Key:</label>
        <input
          type="text"
          value={privateKey}
          onChange={(e) => setPrivateKey(e.target.value)}
        />
      </div>
      <div>
        <label>Public Key:</label>
        <input
          type="text"
          value={publicKey}
          onChange={(e) => setPublicKey(e.target.value)}
        />
      </div>
      <div>
        <label>Signature:</label>
        <input
          type="text"
          value={signature}
          onChange={(e) => setSignature(e.target.value)}
        />
      </div>
      <div>
        <button onClick={handleGenerateKeyPair}>Generate Key Pair</button>
        <button onClick={handleEncryptData}>Encrypt Data</button>
        <button onClick={handleDecryptData}>Decrypt Data</button>
        <button onClick={handleSignData}>Sign Data</button>
        <button onClick={handleVerifySignature}>Verify Signature</button>
        <button onClick={handleHashData}>Hash Data</button>
      </div>
      <div>
        <h3>Result:</h3>
        <p>{result}</p>
      </div>
    </div>
  );
};

export default WalletCryptoGui;
